util/mock/context.go

```diff
@@ -61,9 +61,16 @@ type Context struct {
 
 type wrapTxn struct {
 	kv.Transaction
+	prepared bool
```

Comment:
Can we store the `oracle.Future` here instead `prepared` such that we can get a "real" transaction when call `Wait`.